### PR TITLE
feat: extends use of `observerId` to relevant adapter calls

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: packages/react/src/environments.ts [hunk 1]
```diff
@@ -20,6 +23,7 @@ export const production: EnvironmentConfig = {
     maxIndexingWaitTime: 120000,
     maxMiningWaitTime: 60000,
   },
+  handleResolver: (handle) => `${handle}.lens`,
```
Comment (krzysu) [inline packages/react/src/environments.ts:26]: 👍

File: packages/react/src/profile/adapters/ProfileGateway.ts [hunk 2]
```diff
@@ -6,21 +6,21 @@ import {
...
 
   async getAllProfilesByOwnerAddress(address: string): Promise<Profile[]> {
     const { data } = await this.apolloClient.query<
       GetAllProfilesByOwnerAddressQuery,
       GetAllProfilesByOwnerAddressQueryVariables
     >({
       query: GetAllProfilesByOwnerAddressDocument,
-      variables: { address, limit: 10, sources: this.sources },
+      // 'sources' and 'observerId' are not needed. We just use 'id' and 'handle' for now.
+      variables: { address, limit: 10 },
```
Comment (krzysu) [inline packages/react/src/profile/adapters/ProfileGateway.ts:23]: I think sources are not used in any profile queries anyway, are they?

File: packages/react/src/transactions/infrastructure/ProfileCacheManager.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    const { data } = await this.client.query<GetProfileQuery, GetProfileQueryVariables>({
+      query: GetProfileDocument,
+      variables: {
+        request: {
+          profileId: args.id,
+          handle: args.handle,
+        },
+        observerId: activeProfile?.id,
+        sources: this.sources,
```
Comment (krzysu) [inline packages/react/src/transactions/infrastructure/ProfileCacheManager.ts:40]: do we really need sources to fetch a profile?

Comment (cesarenaldi) [reply packages/react/src/transactions/infrastructure/ProfileCacheManager.ts:40]: It's to get these stats counting correctly: https://github.com/lens-protocol/lens-sdk/pull/229#discussion_r1137114156

**Pull request merged**